parquet-hadoop/src/main/java/parquet/hadoop/CorruptRecordCounter.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package parquet.hadoop;
+
+
+import org.apache.hadoop.conf.Configuration;
+
+import parquet.Log;
+import parquet.io.ParquetDecodingException;
+import parquet.io.api.RecordMaterializer.CorruptRecordException;
+
+// Essentially taken from:
+// https://github.com/twitter/elephant-bird/blob/master/core/src/main/java/com/twitter/elephantbird/mapreduce/input/LzoRecordReader.java#L124
+
+/**
+ * Tracks number of decoding errors and throws ParquetDecodingException
+ * if the rate of errors crosses a limit.<p> These types of errors are meant
+ * to be recoverable record conversion errors, such as a union missing a value, or schema
+ * mismatch and so on. It's not meant to recover from corruptions in the parquet
+ * columns themselves.
+ *
+ * The intention is to skip over very rare file corruption or bugs where
+ * the write path has allowed invalid records into the file, but still catch large
+ * numbers of failures. Not turned on by default (by default, no errors are tolerated).
+ */
+public class CorruptRecordCounter {
+
+  /* Tolerated percent bad records */
+  public static final String BAD_RECORD_THRESHOLD_CONF_KEY = "parquet.read.bad.record.threshold";
+
+  /* Error out only after threshold rate is reached and we have seen this many errors */
+  public static final String BAD_RECORD_MIN_COUNT_CONF_KEY = "parquet.read.bad.record.min";
+
+  private static final Log LOG = Log.getLog(CorruptRecordCounter.class);
+
+  private static final float DEFAULT_THRESHOLD =  0f;
+  private static final long DEFAULT_MIN_ERRORS = 0;
+
+  private long numRecords;
+  private long numErrors;
+
+  private final double errorThreshold; // max fraction of errors allowed
+  private final long minErrors; // throw error only after this many errors
```

Comment:
The idea is, you can set both a percent threshold, and also an absolute minimum. Now that you point this out -- this isn't really needed. 

Elephantbird does this because elephantbird doesn't know the total number of records, so it divides the number of errors by the current record count, which is why they need this minErrors. But we don't even need that, nor the incRecords() method at all, so I think we can simply this by just removing the record count, incRecords() and the minErrors all together